pkg/provider/helm/values_test.go

```diff
@@ -313,6 +441,726 @@ func TestHelmValuesGenerator_GenerateValues_failure(t *testing.T) {
 	}
 }
 
+func TestGetNestedMap(t *testing.T) {
+	tests := []struct {
+		name   string
+		values map[string]any
+		key    string
+		want   map[string]any
+		exists bool
+	}{
+		{
+			name: "map exists, valid key",
+			values: map[string]any{
+				"spire-server": map[string]any{
+					"fullnameOverride": "spire-server",
+				},
+			},
+			key: "spire-server",
+			want: map[string]any{
+				"fullnameOverride": "spire-server",
+			},
+			exists: true,
+		},
+		{
+			name: "map doesn't exist, valid key",
+			values: map[string]any{
+				"spire-server": map[string]any{
+					"caKeyType": "rsa-2048",
+					"caTTL":     "12h",
+				},
+			},
+			key: "global",
+			want: map[string]any{
+				"spire": map[string]any{
+					"clusterName": "local1",
+				},
+			},
```

Comment:
nit: want doesn't make sense here.

Reply:
Maybe we can change all references of `want` to `expected`?

Reply:
Sorry, I wasn't clear. The word want is commonly used in go testing, but it's unused here because this is an error test case.

Reply:
Okay I'll remove it in this test case, tahnks!